docs/glossary: Add decoupled Drupal glossary entry. (#21243)
gatsbyjs/gatsby: docs/docs/glossary/decoupled-drupal.md

When a visitor requests a URL, Drupal retrieves the requested content from the database and merges it with the appropriate template to create an HTML response. The downside of such an architecture is that your site's content is only available as HTML. HTML limits where and how your content can be used.

In a decoupled architecture, Drupal's only responsibility is content. Rather than returning HTML documents, Drupal returns JSON from its [REST](https://www.drupal.org/docs/8/api/restful-web-services-api/restful-web-services-api-overview) or [JSON:API](https://www.drupal.org/docs/8/modules/jsonapi/api-overview) interfaces. Or you can use [GraphQL](/docs/glossary/graphql) by installing the [GraphQL module](https://www.drupal.org/docs/8/modules/graphql).

A decoupled Drupal architecture offers two key advantages over a tightly coupled one.

- **You can use one content management system to serve multiple frontends** &mdash; for example, your Gatsby site, your mobile application, and your smart TV application.
- **You can develop, change, and upgrade the frontend and backend independently of each other. **Upgrading Drupal doesn't require you to modify your site's appearance.

> To use Drupal as content source for Gatsby, add the [`gatsby-source-drupal`](/packages/gatsby-source-drupal/) plugin to your project. As with Gatsby, you install the `gatsby-source-drupal` plugin using [npm](/docs/glossary/#npm).

Reviewer: ```suggestion
To use Drupal as a content source for Gatsby, add the [`gatsby-source-drupal`](/packages/gatsby-source-drupal/) plugin to your project. As with Gatsby itself, you install the `gatsby-source-drupal` plugin using [npm](/docs/glossary/#npm).
```

```shell
npm install --save gatsby-source-drupal
```

A decoupled Drupal architecture lets you use the full power of Drupal's content management tools while gaining the performance advantages of a static Gatsby site.

## Learn more about decoupled Drupal

- Watch [Kyle Mathews’ presentation on Gatsby + Drupal](https://2017.badcamp.net/session/coding-development/beginner/headless-drupal-building-blazing-fast-websites-reactgatsbyjs)

> - [Drupal-GraphQL](https://drupal-graphql.gitbook.io/graphql/) at Gitbook

Reviewer: ```suggestion
- Documentation for the [GraphQL Drupal module](https://drupal-graphql.gitbook.io/graphql/)
```

The fact that they used GitBook to make this doesn't seem super important to me. I think this adjustment makes it clearer.

> - [Sourcing from Drupal](https://www.gatsbyjs.org/docs/sourcing-from-drupal/) in the Gatsby docs

Reviewer: ```suggestion
- [Sourcing from Drupal](/docs/sourcing-from-drupal/) in the Gatsby docs
```
